include/deal.II/differentiation/ad/sacado_product_types.h

```diff
@@ -83,6 +83,31 @@ namespace internal
     typedef Sacado::Fad::DFad<typename ProductType<T,U>::type> type;
   };
 
+
+  // Sacado::Fad::Dfad expression templates
+  // We demote the result of the expression template operations
+  // to a Sacado::Fad::Dfad itself. This is the only way to retain
+  // consistency between the number type going into a complex chain of
+  // (potentially branching) operations and that coming out of them.
```

Comment:
that -> then

Reply:
Ok to merge if the tester is happy.

Reply:
Thanks for looking at this! This is actually correct - one can read it as "between the number type going into... at that (number type) coming out of them".